includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -2088,20 +2089,34 @@ protected function request( $params, $api, $method, $is_site_specific = true, $u
 			);
 		}
 
-		$response = $this->http_client->remote_request(
-			[
-				'url'             => $url,
-				'method'          => $method,
-				'headers'         => apply_filters( 'wcpay_api_request_headers', $headers ),
-				'timeout'         => self::API_TIMEOUT_SECONDS,
-				'connect_timeout' => self::API_TIMEOUT_SECONDS,
-			],
-			$body,
-			$is_site_specific,
-			$use_user_token
-		);
+		$headers        = apply_filters( 'wcpay_api_request_headers', $headers );
+		$stop_trying_at = time() + self::API_TIMEOUT_SECONDS;
```

Comment:
Are there any parts of the request pointing out its time?

Reply:
I'm not sure I understand the question.

Reply:
`time()` returns a timestamp, right?

With testing in mind (we got used to using filters for injecting time in tests), I wonder if we have to use `time()` call here or perhaps there is a timestamp already in the method scope?

Reply:
Gotcha, marking as resolved.